.github/workflows/tests.yml

```diff
@@ -64,71 +63,137 @@ jobs:
+              core.setOutput('test-bot-fail-fast', '')
             }
 
             if (label_names.includes('CI-test-bot-skip-recursive-dependents')) {
               console.log('CI-test-bot-skip-recursive-dependents label found. Passing --skip-recursive-dependents to brew test-bot.')
-              console.log('::set-output name=test-bot-skip-recursive-dependents::--skip-recursive-dependents');
+              core.setOutput('test-bot-skip-recursive-dependents', '--skip-recursive-dependents')
             } else {
               console.log('No CI-test-bot-skip-recursive-dependents label found. Not passing --skip-recursive-dependents to brew test-bot.')
-              console.log('::set-output name=test-bot-skip-recursive-dependents::');
+              core.setOutput('test-bot-skip-recursive-dependents', '')
+            }
+
+            /* RUNNERS */
+
+            const runners = ["11-arm64", "11", "10.15", "10.14"]
+
+            if (label_names.includes('CI-linux-self-hosted')) {
+              runners.push('linux-self-hosted-1')
+            } else {
+              runners.push('ubuntu-latest')
             }
 
+            core.setOutput('runners', JSON.stringify(runners))
+
   tests:
-    needs: tap_syntax
-    if: github.event_name == 'pull_request' && needs.tap_syntax.outputs.syntax-only == 'false'
+    needs: pre_tests
+    if: github.event_name == 'pull_request' && needs.pre_tests.outputs.syntax-only == 'false'
     strategy:
       matrix:
-        version: ["11-arm64", "11", "10.15", "10.14"]
+        runner: ${{fromJson(needs.pre_tests.outputs.runners)}}
       fail-fast: false
-    runs-on: ${{ matrix.version }}
+    runs-on: ${{matrix.runner}}
     timeout-minutes: 4320
     env:
-      PATH: "/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
-      HOMEBREW_DEVELOPER: 1
-      GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED: 1
       GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
       HOMEBREW_GITHUB_API_TOKEN: ${{secrets.GITHUB_TOKEN}}
     steps:
+      - name: Set environment variables
+        run: |
+          if [ "$RUNNER_OS" = 'macOS' ]; then
+            echo 'PATH=/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin' >> $GITHUB_ENV
+            echo 'GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED=1' >> $GITHUB_ENV
+          else
+            echo 'HOMEBREW_CORE_GIT_REMOTE=${{github.event.repository.html_url}}' >> $GITHUB_ENV
+            echo 'HOMEBREW_FORCE_HOMEBREW_ON_LINUX=1' >> $GITHUB_ENV
+          fi
+
       - name: Check whether ARM tests will be forced
-        if: needs.tap_syntax.outputs.force-arm == 'false'
+        if: runner.os == 'macOS' && needs.pre_tests.outputs.force-arm == 'false'
         run: echo 'HOMEBREW_SKIP_UNBOTTLED_ARM_TESTS=1' >> $GITHUB_ENV
 
+      - name: Check whether Linux tests will be forced
+        if: runner.os == 'Linux' && needs.pre_tests.outputs.force-linux == 'false'
+        run: echo 'HOMEBREW_SKIP_UNBOTTLED_LINUX_TESTS=1' >> $GITHUB_ENV
```

Comment:
Could these be combined into a single step which checks `needs.pre_tests.outputs.force-linux` inline and always runs? Maybe not, no worries if not.

Reply:
Should be doable, but why?

Reply:
Just to avoid more steps which make the output longer and having a guaranteed skipped step in every circumstance.